core/src/main/java/hudson/FilePath.java

```diff
@@ -1409,7 +1410,7 @@ public FilePath createTempDir(final String prefix, final String suffix) throws I
                 public String invoke(File dir, VirtualChannel channel) throws IOException {
                     File f = File.createTempFile(prefix, suffix, dir);
                     f.delete();
-                    f.mkdir();
+                    Files.createTempDirectory(f.toPath(), null);
```

Comment:
This function is being updated by #3161, so I'd remove this change as it will cause a merge conflict.